shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/BridgeIngressServiceImpl.java

```diff
@@ -83,13 +85,22 @@ public void createBridgeIngress(BridgeDTO bridgeDTO) {
 
     @Override
     public void deleteBridgeIngress(BridgeDTO bridgeDTO) {
-        final String namespace = customerNamespaceProvider.resolveName(bridgeDTO.getCustomerId());
-        final boolean bridgeDeleted =
-                kubernetesClient
-                        .resources(BridgeIngress.class)
-                        .inNamespace(namespace)
-                        .delete(BridgeIngress.fromDTO(bridgeDTO, namespace));
-        if (!bridgeDeleted) {
+        String customerId = bridgeDTO.getCustomerId();
+        String bridgeId = bridgeDTO.getId();
+
+        // Deletion of the Namespace will cascade deletion of the BridgeIngress...
+        // In a *real* k8s environment the Namespace deletion should cascade into all children.
+        final String namespace = customerBridgeNamespaceProvider.resolveName(customerId, bridgeId);
```

Comment:
@manstis The alternative being that we move the delete of the `namespace` into the `BridgeIngressController.cleanUp()` method. That leaves us with a single, consistent code-path regardless of environment.

If we're going to rely on the cascade, then we definitely need to verify that it works as expected. In particular I'm looking at https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/main/shard-operator-integration-tests/src/test/java/com/redhat/service/smartevents/shard/operator/cucumber/steps/BridgeIngressSteps.java#L62-L69 

I suspect this will need updating anyway as it relies on the namespace being available.

Reply:
@manstis We also need to be careful here in that in our existing environments we have namespaces with multiple bridges in them. So this will delete all those bridges rather than just the single bridge. So we need to consider a strategy for the situation where there are multiple bridges in a `namespace`.